modules/olm-arch-catalog-operator.adoc

```diff
@@ -0,0 +1,25 @@
+// Module included in the following assemblies:
+//
+// * operators/understanding_olm/olm-arch.adoc
+
+[id="olm-arch-catalog-operator_{context}"]
+= Catalog Operator
+
+The Catalog Operator is responsible for resolving and installing CSVs and the
+required resources they specify. It is also responsible for watching
+CatalogSources for updates to packages in channels and upgrading them
+(optionally automatically) to the latest available versions.
+
+A user who wishes to track a package in a channel creates a Subscription
+resource configuring the desired package, channel, and the CatalogSource from
+which to pull updates. When updates are found, an appropriate InstallPlan is
+written into the namespace on behalf of the user.
+
+.Catalog Operator workflow
+* Has a connection to each CatalogSource in the cluster.
```

Comment:
I feel like this needs an intro sentence. Reading from the subhead to the list is not something I'm used to seeing.
Even just:

> With the Catalog Operator workflow, the Operator:

Also, instead of "Has a connection", can you write, "Connects to each..."?